lib/client/https_client.go

```diff
@@ -101,7 +102,7 @@ func (w *WebClient) PostJSONWithFallback(ctx context.Context, endpoint string, v
 	// If we're not allowed to try plain HTTP, bail out with whatever error we have.
 	// Note that we're only allowed to try plain HTTP on the loopback address, even
 	// if the caller says its OK
-	if !(allowHTTPFallback && apiutils.IsLoopback(u.Host)) {
+	if !(allowHTTPFallback) {
```

Comment:
This should inspect the proxy to see if it's a local address. Unfortunately I couldn't determine how to identify the proxy the request would use given the APIs available here.

Reply:
By proxy here, you want to check if you have a local `HTTP_CONNECT` proxy that you are connecting through right? So you want a similar block to what you had above right?

```
	if !(allowPlainHTTP && (utils.IsLoopback(req.URL.Host) || utils.IsLoopback(proxy.Host))) {
```

Reply:
Yes, but I wasn't actually able to determine a way to find the proxy used for this request within this function.